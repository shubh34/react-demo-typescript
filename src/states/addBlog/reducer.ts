import { IAddBlogState } from './types';
import { INIT_ADD_BLOG, UPDATE_FORM_FIELD, ADD_BLOG_CATEGORY, DELETE_BLOG_CATEGORY } from './actions';


const addBlogInitialState: IAddBlogState = {
   title:  '',
   categories: [],
   content: '',
}



const reducer = (state = addBlogInitialState, action: any): IAddBlogState => {
	switch (action.type) {
    case INIT_ADD_BLOG : 
        return {
            ...state,
            ...addBlogInitialState,
        }
    case UPDATE_FORM_FIELD: 
        return {
            ...state,
            [action.field.name]: action.field.value,
        }
    case ADD_BLOG_CATEGORY: 
        return {
            ...state,
            categories: [...state.categories, action.category]
        }
    case DELETE_BLOG_CATEGORY: 
        return {
            ...state,
            categories: state.categories.filter(category => category !== action.category)
        }
	default:

		return state;
	}
};
export default reducer;
