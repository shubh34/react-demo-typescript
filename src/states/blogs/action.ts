import uuid from 'node-uuid';
import moment from 'moment';
import { IBlog } from "./types";

export const ADD_BLOG: string = "ADD_BLOG";
export const EDIT_BLOG: string = "EDIT_BLOG";
export const DELETE_BLOG: string = "DELETE_BLOG";
export const VIEW_BLOG: string = "VIEW_BLOG";

export const addBlog = (blog: IBlog) => {
    return {
        type: ADD_BLOG,
        blog: {
            ...blog,
            id: uuid.v1(),
            date: moment(),
        },
    }
}

export const deleteBlog = (blogId: string) => {
    return {
        type: DELETE_BLOG,
        blogId: blogId,
    }
}

export const ediBlog = (blog: IBlog) => {
    return {
        type: EDIT_BLOG,
        blog,
    }
}




