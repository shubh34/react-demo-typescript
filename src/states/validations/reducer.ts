import { IValidationState } from './types';
import { RESET_VALIDATION, INIT_VALIDATION, UPDATE_VALIDATION } from './actions';

const initialValidation: IValidationState = {
    form: {
        title: {
            isValid :  false, 
        },
        categories: {
            isValid :  false, 
        },
        content: {
            isValid: false,
        }
    },
}



const reducer = (state = initialValidation, action: any): IValidationState => {
	switch (action.type) {
    case RESET_VALIDATION : 
        return {
            ...state,
            ...initialValidation,
        }
    case INIT_VALIDATION: 
        return {
            ...state,
            form: {
                ...state.form,
                [action.field.name]: {
                    isValid: action.field.isValid,
                }
            }
        }
    case UPDATE_VALIDATION: 
        return {
            ...state,
            form: {
                ...state.form,
                [action.field.name]: {
                    isValid: action.field.isValid,
                }
            }
        }
     
	default:
		return state;
	}
};
export default reducer;
