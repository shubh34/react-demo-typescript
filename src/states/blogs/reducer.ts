import { IBlogsState } from "./types";
import { ADD_BLOG } from "./action";

const init: IBlogsState = {
    blogs: []
};


const reducer = (state = init, action: any) : IBlogsState => {
	switch (action.type) {
	case ADD_BLOG: 
		return {
			...state,
			blogs: [...state.blogs, action.blog],
		}
	default:
		console.log('state', state);
		return state;
	}
};
export default reducer;
