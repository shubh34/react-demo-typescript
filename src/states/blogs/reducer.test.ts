import reducer from  './reducer';
import moment from 'moment';

import { addBlog } from './action';

export const state = {
        blogs: [{
             id: '15',
             categories: ['Food', 'Travel'],
             date: moment('21 Jan 2019').subtract(2, 'days'),
             title: 'Food Blog',
             content: 'Lorem ipsum .'
         },
     ]
}
describe('Reducer selector', () => {
    it('should add a blog on addBlog action', () => {
        const newBlog ={
            id: '1122',
            categories: ['Food', 'Travel'],
            date: moment('21 Jan 2019').subtract(2, 'days'),
            title: 'Food Blog',
            content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempo.'
        };
       expect(reducer(state, addBlog(newBlog))).toMatchSnapshot();
    });
})