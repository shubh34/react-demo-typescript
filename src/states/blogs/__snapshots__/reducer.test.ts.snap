// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer selector should add a blog on addBlog action 1`] = `
Object {
  "blogs": Array [
    Object {
      "categories": Array [
        "Food",
        "Travel",
      ],
      "content": "Lorem ipsum .",
      "date": "2019-01-19T00:00:00.000Z",
      "id": "15",
      "title": "Food Blog",
    },
    Object {
      "categories": Array [
        "Food",
        "Travel",
      ],
      "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempo.",
      "date": "2019-01-19T00:00:00.000Z",
      "id": "1122",
      "title": "Food Blog",
    },
  ],
}
`;
